include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h

```diff
@@ -226,13 +256,19 @@ struct __subgroup_radix_sort
 #pragma unroll
                             for (uint16_t __i = 0; __i < __block_size; ++__i)
                             {
+                                if constexpr (!__uniform_space)
+                                {
+                                    const uint16_t __idx = __wi * __block_size + __i;
+                                    if(__idx >= __n)
+                                        continue;
+                                }
                                 const uint16_t __r = __indices[__i];
                                 if (__r < __n)
-                                    __src[__r] = __keys[__i];
+                                    __src[__r] = __vals[__i];
                             }
                             return;
                         }
-                        __to_blocked(__it, __wi, __exchange_lacc, __keys, __indices);
+                        __to_blocked<__uniform_space>(__it, __wi, __exchange_lacc, __vals, __n, __indices);
```

Comment:
Let's discuss `__uniform_space`. I don't currently understand what's going on. Having that understating we could come up to better solution.

Reply:
Lets discuss it offline (how I suggested)

Reply:
this comment is not actual any more due to the code was re-written another way.